lib/ex_unit/lib/ex_unit/capture_log.ex

```diff
@@ -0,0 +1,133 @@
+defmodule ExUnit.CaptureLog do
+  @moduledoc ~S"""
+  Functionality to capture logs for testing.
+
+  ## Examples
+
+      defmodule AssertionTest do
+        use ExUnit.Case
+
+        import ExUnit.CaptureLog
+
+        test :example do
+          assert capture_log(fn ->
+            Logger.error "log msg"
+          end) =~ "log msg"
+        end
+
+        test "check multiple captures concurrently" do
+          fun = fn ->
+            for msg <- ["hello", "hi"] do
+              assert capture_log(fn -> Logger.error msg end) =~ msg
+            end
+            Logger.debug "testing"
+          end
+          assert capture_log(fun) =~ "testing"
+        end
+      end
+
+  """
+
+  @doc """
+  Captures Logger messages generated when evaluating `fun`.
+
+  Returns the binary which is the captured output.
+
+  By default, `capture_log` replaces the `group_leader` (`:stdio`)
+  for the current process and captures any log messages sent by
+  processes wth the replacement `group leader`. However, the
+  capturing of log messages with a `group_leader` of any other
+  device is also possible by giving the device explicitly as an
+  argument. It is also possible to capture all log messages by
+  giving `nil` for the device.
+
+  Note that when capturing something other than `:stdio`, log
+  messages from other tests or processes might be captured. However
+  it is safe to use async true if this can be handled. Multiple
+  captures (and other backends) can handle log messages at the same
+  time. For example the `:console` backend will still log messages
+  to the console.
+
+  It is possible to capture all log messages by giving `nil` for
+  the device.
+
+  It is possible to configure the level to capture with `:level`,
```

Comment:
Ok. I think we can remove this paragraph then because it is not really about capture_log but if you are calling any of the Logger functions. And what would be true for any other configuration too. :D